astrosat/utils/utils_logging.py

```diff
@@ -53,10 +52,21 @@ def emit(self, record):
                 ],
             )
 
-            db_record = DatabaseLogRecord.objects.create(
-                logger_name=record.name,
-                level=record.levelno,
-                message=record.getMessage(),
-                trace=trace,
-            )
+            try:
+                uuid = getattr(record, 'uuid')
+
+                db_record = DatabaseLogRecord.objects.create(
+                    logger_name=record.name,
+                    level=record.levelno,
+                    message=record.getMessage(),
+                    uuid=uuid,
+                    trace=trace,
+                )
+            except AttributeError:
+                db_record = DatabaseLogRecord.objects.create(
```

Comment:
Why is this repeated in a try/catch block.  Is it b/c of the case where "uuid" wasn't included in the `extra` arguments and therefore `getattr` failed?

If so, `getattr` takes a default argument to return if the attribute doesn't exist on the object:

```
getattr(object, "attribute", default)
```

Reply:
You guessed, `getattr` was constantly failing for tests with no `uuid` in the `extra` block. Your point here kinda answers my previous response. Taking this and your previous comment, maybe we should do `getattr(record, 'uuid', uuid.uuid4())`, that way there is never a `None` entry and we just have the one:

```python
db_record = DatabaseLogRecord.objects.create(
    logger_name=record.name,
    level=record.levelno,
    message=record.getMessage(),
    uuid=uuid,
    trace=trace,
)
```

bit of code. That would cover both `orbis` and `covid-app` usages I think, but please correct me if I'm wrong.